flax/linen/module.py

```diff
@@ -407,52 +511,60 @@ def __post_init__(self):
     self.setup()
 
   def __repr__(self):
-    return module_repr(self)
+    return _module_repr(self)
 
   def setup(self):
-    """Called when Module instance receives variables and PRNGs.
-    If you want to use PRNGs and/or read or write variables during module
-    instance construction, override this method. It will be automatically
-    called from the dataclass `__post_init__`.
+    """Initializes a Module (similar to ``__init__`` for non-dataclass Python classes).
+
+    Override this method in your module subclasses to initialize submodules and
+    other attributes. This method is called after all dataclass attributes are
+    assigned and the module is ready for use. Variables and RNGs are guaranteed
+    to be available.
```

Comment:
what do you want to convey with this phrasing?

Reply:
I wrote this comment originally. Maybe we can do better but the point is that `setup` acts like `__init__` but not ... exactly. In this code:

```
class MyModule(nn.Module):
  def setup(self):
    self.submodule = MySubModule()
```

Then `setup` is only called after the assignment (because then we know the attribute name hence the RNG and the place parameters live in the variable dict).

So I was trying to somehow explain that. But maybe I did a poor job. Maybe instead of explaining the technicalities I should just say exactly when setup is called, like I explained in this comment.

Reply:
By "the assignment" I mean the line `self.submodule = MySubModule()` [since we override `__setattr__`]

Reply:
I've updated the comment!